gnomad_qc/v4/create_release/create_combined_faf_release_ht.py

```diff
@@ -152,22 +157,115 @@ def extract_freq_info(
     return ht
 
 
+def add_all_sites_an_and_qual_hists(
+    ht: hl.Table,
+    exomes_all_sites_ht: hl.Table,
+    genomes_all_sites_ht: hl.Table,
+) -> hl.Table:
+    """
+    Add all sites AN and qual hists to the Table.
+
+    :param ht: Table with frequency and FAF information.
+    :param exomes_all_sites_ht: Table with all sites AN and qual hists for exomes.
+    :param genomes_all_sites_ht: Table with all sites AN and qual hists for genomes.
+    :return: Table with all sites AN and qual hists added.
+    """
+
+    def _get_freq_from_an_expr(
+        freq_expr: hl.expr.ArrayExpression,
+        all_sites_an_expr: hl.expr.ArrayExpression,
+        freq_meta_expr: hl.expr.ArrayExpression,
+        an_meta_expr: hl.expr.ArrayExpression,
+    ) -> hl.expr.ArrayExpression:
+        """
+        Get freq expression with all sites AN added if `freq_expr` is missing.
+
+        :param freq_expr: ArrayExpression of frequencies.
+        :param all_sites_an_expr: ArrayExpression of all sites AN.
+        :param freq_meta_expr: Frequency metadata.
+        :param an_meta_expr: AN metadata.
+        :return: ArrayExpression of frequencies with all sites AN added if `freq_expr`
+            is missing.
+        """
+        meta_map = hl.dict(hl.enumerate(an_meta_expr).map(lambda x: (x[1], x[0])))
+        all_sites_an_expr = freq_meta_expr.map(
+            lambda x: hl.struct(
+                AC=0,
+                AF=hl.missing(hl.tfloat64),
```

Comment:
Wouldn't we want to fill this in with 0 where we can? I saw there are 16k sites in the exomes AN where AN raw is 0 so can't set it to 0 but maybe we should compute here?

Reply:
I guess the AN would be zero in many groups at many sites, not just raw..

Reply:
Ah, yes good point, thank you